File: Ass2.4.js
function ChkAmstrong(No) {
    var Value = No;
    var Temp = No;
    var Cnt = 0;
    var Ans = 0;
    var Falg = false;
    var REM = 0;
    var Sum = 1;
    var Counter = 0;
    while (Value != 0) {
        Value = Math.floor(Value / 10);
        Cnt++;
    }
    while (Temp != 0) {
        REM = Temp % 10;
        while (Counter < Cnt) {
            Sum = Sum * REM;
            Counter++;
        }
        Ans = Ans + Sum;
        Temp = Math.floor(Temp / 10);
        Sum = 1;
        Counter = 0;
    }
    if (No == Ans) {
        Falg = true;
    }
    return Falg;
}
var Ret = false;
Ret = ChkAmstrong(9444);
if (Ret == true) {
    console.log("It is Amstrong Number");
}
else {
    console.log("It is not Amstrong Number");
}

File: Ass2.4.py

def FactSum(No):
    Sum = 0
    for i in range(1,int(No/2)+1):
        if(No % i == 0):
            Sum = Sum + i
    return Sum
    
def main():
    Value = int(input("Enter number : "))

    Ans = FactSum(Value)
    print("Addition of factorial is : ",Ans)

if __name__ == "__main__":
    main()